src/Microsoft.Owin.Security.Authorization.Mvc/ResourceAuthorizeAttribute.cs

```diff
@@ -45,11 +45,12 @@ protected override bool AuthorizeCore(HttpContextBase httpContext)
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            var controller = httpContext.Items[s_controllerKey] as IAuthorizationController;
+            var filterContext = (System.Web.Mvc.AuthorizationContext)httpContext.Items[s_authorizationContextKey];
```

Comment:
This should be soft casted. Hard casting will throw an InvalidCastException if someone overrides the stored item externally.
```suggestion
            var filterContext = httpContext.Items[s_authorizationContextKey] as System.Web.Mvc.AuthorizationContext;
```